frontend/app/src/components/settings/controls/SettingsOption.vue

```diff
@@ -57,24 +87,61 @@ export default defineComponent({
       const func = get(transform);
       const settingKey = get(setting);
       const settingValue = func ? func(newValue) : newValue;
-      const result = await updateSetting(
-        {
-          [settingKey]: settingValue
-        },
-        {
-          success: getMessage(successMessage, newValue),
-          error: getMessage(errorMessage, newValue)
+      const validators = get(rules);
+
+      let failed: boolean = false;
+
+      validators.forEach(validator => {
+        const validate = validator(settingValue);
+        if (validate !== true) {
+          setError(validate as string, false);
+          failed = true;
+          return;
         }
-      );
+      });
+
+      if (failed) {
+        emit('finished');
+        return;
+      }
+
+      let result;
+
+      if (get(sessionSetting)) {
```

Comment:
Can we wrap this and create a function that lives in the settings composable?

Reply:
Hmm I'm not sure I get it, which part do you want to move to composable?

Reply:
Sorry about this, I didn't properly select the range for the comment and I couldn't find a way to edit it afterward. 

I meant the whole update `if/else` from `110` to `134`